retrofit-converters/wire/src/main/java/retrofit/converter/WireConverter.java

```diff
@@ -36,31 +36,24 @@ public WireConverter(Wire wire) {
       throw new IllegalArgumentException("Expected a proto message but was " + c.getName());
     }
 
-    if (!MIME_TYPE.equalsIgnoreCase(body.mimeType())) {
-      throw new IllegalStateException("Expected a proto but was: " + body.mimeType());
```

Comment:
2 things:
I don't mind dropping this check.
It should probably be a separate PR. (So it can be rolled-back or referenced independently).